K2Field.K2NE.ServiceBroker/ServiceObjects/ExcelServices/ExcelImportServiceSO.cs

```diff
@@ -0,0 +1,528 @@
+            mUploadExcelDataToASmartObject.ReturnProperties.Add(Constants.SOProperties.ExcelImportServices.Results);
+            
+            mUploadExcelDataToASmartObject.InputProperties.Add(Constants.SOProperties.ExcelImportServices.ExcelFile);
+            mUploadExcelDataToASmartObject.Validation.RequiredProperties.Add(Constants.SOProperties.ExcelImportServices.ExcelFile);
+            mUploadExcelDataToASmartObject.InputProperties.Add(Constants.SOProperties.ExcelImportServices.SheetName);
+            mUploadExcelDataToASmartObject.InputProperties.Add(Constants.SOProperties.ExcelImportServices.SmartObject);
+            mUploadExcelDataToASmartObject.Validation.RequiredProperties.Add(Constants.SOProperties.ExcelImportServices.SmartObject);
+            mUploadExcelDataToASmartObject.InputProperties.Add(Constants.SOProperties.ExcelImportServices.CreateMethodName);            
+            mUploadExcelDataToASmartObject.InputProperties.Add(Constants.SOProperties.ExcelImportServices.HeaderRowSpaces);
+            mUploadExcelDataToASmartObject.InputProperties.Add(Constants.SOProperties.ExcelImportServices.TransactionIDName);
+            mUploadExcelDataToASmartObject.InputProperties.Add(Constants.SOProperties.ExcelImportServices.TransactionIDValue);
+            mUploadExcelDataToASmartObject.InputProperties.Add(Constants.SOProperties.ExcelImportServices.FQN);
+            
+            so.Methods.Add(mUploadExcelDataToASmartObject);
+            
+            soList.Add(so);
+
+            return soList;
+        }
+
+        public override void Execute()
+        {
+            switch (ServiceBroker.Service.ServiceObjects[0].Methods[0].Name)
+            {
+                case Constants.Methods.ExcelImportServices.UploadExcelDataToASmartObject:
+                    UploadExcelDataToASmartObject();
+                    break;
+            }
+        }
+
+        private void UploadExcelDataToASmartObject()
+        {
+            //Get input properties
+            FileProperty excelFile = GetFileProperty(Constants.SOProperties.ExcelImportServices.ExcelFile, true);
+            string smartObject = GetStringProperty(Constants.SOProperties.ExcelImportServices.SmartObject, true);
+            string createMethodName = GetStringProperty(Constants.SOProperties.ExcelImportServices.CreateMethodName, false);
+            string headerRowSpaces = GetStringProperty(Constants.SOProperties.ExcelImportServices.HeaderRowSpaces, false);
+            string transactionIDName = GetStringProperty(Constants.SOProperties.ExcelImportServices.TransactionIDName, false);
+            string transactionIDValue = GetStringProperty(Constants.SOProperties.ExcelImportServices.TransactionIDValue, false);
+            string FQN = GetStringProperty(Constants.SOProperties.ExcelImportServices.FQN, false);
```

Comment:
Instead of accepting a FQN as a parameter, use CallingFQN as that will always return the correct value and avoids the user having to provide it.

Reply:
Done.